yarl/_url.py

```diff
@@ -186,18 +229,18 @@ def __new__(cls, val="", *, encoded=False, strict=None):
     def build(
         cls,
         *,
-        scheme="",
-        authority="",
-        user=None,
-        password=None,
-        host="",
-        port=None,
-        path="",
-        query=None,
-        query_string="",
-        fragment="",
-        encoded=False,
-    ):
+        scheme: str = "",
+        authority: str = "",
+        user: Optional[str] = None,
```

Comment:
Please, avoid using `Optional` in favor of `| None` since it's considered less confusing:
```suggestion
        user: str | None = None,
```

Reply:
But, as mentioned before, this breaks runtime type checking on older Python versions, if the user tries to use those types. We can leave this and just let pyupgrade change next year.

Reply:
Yup, we have to keep it for 3.8   Once we drop 3.8, we can use `from __future__ import annotations` to turn it on in 3.9 and than run pyupgrade

Reply:
Still a runtime error on 3.9 if you use get_type_hints() or similar..

Reply:
@Dreamsorcerer don't we already do this elsewhere? Though, if `|` is not an option, a compatible variant would be `Union[str, None]`.

Reply:
#1095 Replace all `Optional[str]` with `Union[str, None]`

Reply:
I think we shouldn't change `aiohttp`. This one is has `Union` and `Optional` types quite a bit so we should go one way or the other for consistency, and since @webknjaz has a strong preference for `Union` I'm fine with that here knowing that they will eventually be the `type | None` syntax I personally much prefer.